crates/telio-firewall/src/firewall.rs

```diff
@@ -460,18 +500,23 @@ impl StatefullFirewall {
         }
     }
 
-    fn handle_outbound_tcp<'a>(&self, ip: &impl IpPacket<'a>) {
-        let (key, flags) = unwrap_option_or_return!(Self::build_tcp_key(ip, false));
+    fn handle_outbound_tcp<'a>(&self, pubkey: PublicKey, ip: &impl IpPacket<'a>) {
+        let (link, packet) = unwrap_option_or_return!(Self::build_tcp_key(ip, false));
+        let key = TcpConn { link, pubkey };
+
+        let flags = packet.get_flags();
+
         let mut tcp_cache = unwrap_lock_or_return!(self.tcp.lock());
 
         if flags & TCP_FIRST_PKT_MASK == TcpFlags::SYN {
-            telio_log_trace!("Inserting TCP conntrack entry {:?}", key);
+            telio_log_trace!("Inserting TCP conntrack entry {:?}", key.link);
             tcp_cache.insert(
                 key,
                 TcpConnectionInfo {
                     tx_alive: true,
                     rx_alive: true,
                     conn_remote_initiated: false,
+                    next_seq: 0,
```

Comment:
Due to the fact that in this specific moment, we do not know what is the actual `next_seq` number, maybe it would be better represented by Option<u32> with the value `None` in this situation? 

While it does not seem to be strictly speaking incorrect, having `TcpConnectionInfo` record in this stage, if `.reset_existing_connections()` gets called during this state, it  would mean, that TCP RST packet will be generated for not yet established connection with incorrect seq number (`0` in our case). I am thinking, that it would be a bit better if we would not trigger `TCP RST` in such case...

Reply:
Done